src/libmongoc/src/mongoc/mongoc-server-description.c

```diff
@@ -1123,6 +1131,14 @@ _mongoc_server_description_equal (mongoc_server_description_t *sd1,
       return false;
    }
 
+   if (0 != memcmp (&sd1->error, &sd2->error, sizeof (bson_error_t))) {
+      return false;
+   }
+
+   if (!bson_equal (&sd1->topology_version, &sd2->topology_version)) {
+      return false;
+   }
+
```

Comment:
The equality comparison for a server description is specified in SDAM:

https://github.com/mongodb/specifications/blob/a25c50976807b4db16f85fd8d5c5b0337cd29cf9/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#serverdescription